recipes/compute.rb

```diff
@@ -61,6 +61,21 @@
     not_if '/sbin/ppc64_cpu --smt 2>&1 | ' \
       'grep -E \'SMT is off|Machine is not SMT capable\''
   end
+when 'x86_64'
+  kernel_module 'kvm_intel' do
+    onboot true
+    reload false
+    options %w(nested=1)
+    check_availability true
+    only_if { node.deep_fetch('dmi', 'processor', 'manufacturer') == 'Intel(R) Corporation' }
+  end
+  kernel_module 'kvm_amd' do
+    onboot true
+    reload false
+    options %w(nested=1)
+    check_availability true
+    only_if { node.deep_fetch('dmi', 'processor', 'manufacturer') == 'AMD' }
+  end
```

Comment:
Since all the parameters are exactly the same between both kernel modules, could the name of the `kernel_module` block be put in a variable that is set outside the `kernel_module` block?

Something like:
```ruby
proc_type = node.deep_fetch('dmi', 'processor', 'manufacturer')
kvm_mod = if proc_type == 'Intel(R) Corporation'
            'kvm_intel'
          elsif proc_type == 'AMD'
            'kvm_amd'
          end

kernel_module kvm_mod do
  onboot true
  reload false
  options %w(nested=1)
  check_availability true
end
```

This is to prevent code duplication, which can make maintenance harder down the road.

Reply:
What happens when it doesn't find anything? I think it's better to leave this as-is.